caluma/caluma_form/models.py

```diff
@@ -202,14 +202,39 @@ def create_document_for_task(self, task, user):
 class Document(UUIDModel):
     objects = DocumentManager()
 
-    family = models.UUIDField(
-        help_text="Family id which document belongs too.", db_index=True
+    family = models.ForeignKey(
+        "self",
+        help_text="Family id which document belongs too.",
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
     )
     form = models.ForeignKey(
         "caluma_form.Form", on_delete=models.DO_NOTHING, related_name="documents"
     )
     meta = JSONField(default=dict)
 
+    def set_family(self, root_doc):
```

Comment:
We now have the mothod `set_family()` and the signal receiver `set_document_family()`. This could be a bit confusing. It would be nice to have a doc string here, that explains this method.

Reply:
Updated the code a bit, added some documentation (and the signal receiver now uses the method as well)